src/codegen/templates/_grpc_interpreter.py.mako

```diff
@@ -159,3 +165,8 @@ def _assign_numpy_array(numpy_array, grpc_array):
         numpy_array.flat[:grpc_array_size] = numpy.frombuffer(grpc_array, dtype=numpy.uint8)
     else:
         numpy_array.flat[:grpc_array_size] = grpc_array
+
+def _validate_array_dtype(numpy_array, expected_numpy_array_dtype):
+    """Raises Typerror if array type doesn't match with expected numpy.dtype"""
+    if expected_numpy_array_dtype != numpy.generic and numpy_array.dtype != expected_numpy_array_dtype:
+        raise errors.DaqError(f"TypeError: array must have data type {expected_numpy_array_dtype}", DAQmxErrors.UNKNOWN)
```

Comment:
This raises `DaqError`, not `TypeError`.
https://docs.python.org/3/library/exceptions.html#TypeError

```suggestion
    """Raises TypeError if array type doesn't match with expected numpy.dtype"""
    if expected_numpy_array_dtype != numpy.generic and numpy_array.dtype != expected_numpy_array_dtype:
        raise TypeError(f"array must have data type {expected_numpy_array_dtype}")
```

Reply:
Removed DaqError & placed TypeError